sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/ApproximatePercentile.scala

```diff
@@ -321,4 +331,22 @@ object ApproximatePercentile {
   }
 
   val serializer: PercentileDigestSerializer = new PercentileDigestSerializer
+
+  def apply(
+      child: Expression,
+      percentageExpression: Expression,
+      accuracyExpression: Expression,
+      mutableAggBufferOffset: Int, inputAggBufferOffset: Int): ApproximatePercentile =
+    new ApproximatePercentile("percentile_approx", child, percentageExpression,
```

Comment:
sorry to bring this up so late, but after more thoughts, I feel it's too invasive to add a `funcName` parameter to expression constructor: we need to add a lot of `apply` methods and constructors to provide the default function name.

This problem wasn't obvious when we were changing `BoolAnd`, but it's a bit obvious now for `ApproximatePercentile` which already have a lot of constructors.

I'd like to propose another idea: using mutable states via `TreeNodeTag`. The idea is pretty simple: an expression can have an optional alias, which is stored in the expression itself via `TreeNodeTag`. We can update `ApproximatePercentile.prettyName` to `getTagValue(THE_TAG).getOrElse("percentile_approx")`.

In `expressionWithAlias`, it can just call `expression`, and set the TreeNodeTag with the alias at the end.

Reply:
It's a great idea. I will implement and update this PR. Thanks @cloud-fan